### PR TITLE
chore(create-ci): setup CI for testing and building

### PR DESCRIPTION
**Pull request by FlorianFlouquet**
# Description
Creation of a file to automate build and test process

# Linked Issues
#17 

# Changes
CI got harsher

# Screenshots
N/A

# Tests
How has it been tested ?

- [x] Manually
- [ ] Automated tests
- [ ] QA
- [ ] Other

# Additional information
N/A

### CODE REVIEW

File: .github/workflows/test_&_build.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        working-directory: yaki_mobile
+    steps: 
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '17.x'
+          distribution: 'temurin'
+      - uses: subosito/flutter-action@v2
+        with:
+          flutter-version: '3.7.3'
```
Comment (PiotrFLEURY) [inline .github/workflows/test_&_build.yaml:17]: latest version is 3.7.6

**Review by PiotrFLEURY — APPROVED** [review]
Just a comment about flutter version, rest is OK

**Review by HaddoucheRym — APPROVED** [review]

File: .github/workflows/test_&_build.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  flutter-web:
+    runs-on: ubuntu-latest
+    defaults: 
+      run:
+        working-directory: yaki_mobile
+    steps:
+      - uses: actions/checkout@v3
+      - uses: subosito/flutter-action@v2
+        with:
+           flutter-version: '3.7.3'
```
Comment (PiotrFLEURY) [inline .github/workflows/test_&_build.yaml:40]:
```suggestion
           flutter-version: '3.7.6'
```

File: yaki_backend/src/__test__/server.test.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
 /* Testing the code. */
 test("it should pass", async () => {
```
Comment (PiotrFLEURY) [inline yaki_backend/src/__test__/server.test.ts:2]: delete this useless test

File: yaki_mobile/lib/presentation/ui/authentication/authentication.dart [hunk 4]
```diff
@@ -3,35 +3,27 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
...
   }
 
-
-
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     bool isChecked = false;
 
-    final providerlogin = ref.watch(loginProvider);
+    ref.watch(loginProvider);
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/presentation/ui/authentication/authentication.dart:26]: useless, remove it

File: yaki_mobile/test/widget_test.dart [hunk 5]
```diff
@@ -5,13 +5,16 @@
 // gestures. You can also use WidgetTester to find child widgets in the widget
 // tree, read text, and verify that the values of widget properties are correct.
 
-import 'package:flutter/material.dart';
+// import 'package:flutter/material.dart';
```
Comment (PiotrFLEURY) [inline yaki_mobile/test/widget_test.dart:8]: remove useless test

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

**Review by MaximeHavez — DISMISSED** [review]

**Review by PiotrFLEURY — APPROVED** [review]

**Review by LaurentMag — APPROVED** [review]

**Pull request merged**